packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -263,6 +263,15 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
       _positionController.forward();
     else
       _positionController.reverse();
+
+    switch(defaultTargetPlatform) {
```

Comment:
Worth double checking but I have a suspicion this isn't right. We should produce the knock when the knob hits the side, not when it starts moving. We should attach it to the _positionController's status handler.

Reply:
Actually seems (unlike the picker) that it bumps immediately on tapUp/dragstart.